src/infra/core/src/use_cases/delete_dataset_use_case_impl.rs

```diff
@@ -89,23 +92,66 @@ impl DeleteDatasetUseCaseImpl {
 
         Ok(())
     }
+
+    async fn ensure_no_out_of_sync_remotes(
+        &self,
+        dataset_handle: &DatasetHandle,
+        force: bool,
+    ) -> Result<(), DeleteDatasetError> {
+        let ds = self.dataset_repo.get_dataset_by_handle(dataset_handle);
+
+        match ds.get_summary(GetSummaryOpts::default()).await {
+            Ok(summary) => match ds.get_push_info().await {
+                Ok(ds_pushes) => {
+                    // TODO: check only repos from config
+                    let head = summary.last_block_hash;
+                    let pushes: Vec<DatasetPush> = ds_pushes
+                        .pushes
+                        .into_iter()
+                        .filter(|(_, v)| v.head != head)
+                        .map(|(_, v)| v)
+                        .collect();
+                    if !pushes.is_empty() {
+                        let targets: Vec<DatasetRefRemote> =
+                            pushes.into_iter().map(|p| p.target).collect();
+                        let err: DeleteDatasetError = OutOfSyncError { targets }.into();
+                        if force {
+                            eprintln!("{}: {}", console::style("Warning").yellow(), err,);
+                            Ok(())
+                        } else {
+                            Err(err)
+                        }
```

Comment:
This is a bad place to display this warning. Rather you need to define a listener interface, as there is no stylish console when you are running an API server.